scripts/newtab.js

```diff
@@ -0,0 +1,188 @@
+            canvas.style.backgroundRepeat = "no-repeat";
+          }
+          img.addEventListener("mouseover", onMouseOver);
+          img.addEventListener("mouseout", onMouseLeave);
+          listener = true;
+        })
+      })
+    } else if (result.newtab == "pics"){
+      document.querySelectorAll(".thumbnail").forEach(function(img){
+        if(listener){
+          img.removeEventListener(onMouseOver);
+          img.removeEventListener(onMouseLeave);
+        }
+        var newImg = document.createElement('img');
+        img.parentNode.replaceChild(newImg, img);
+        let rand = Math.floor(Math.random() * placeholders.length);
+        newImg.src = placeholders[rand];
+        newImg.style.opacity = 0;
+        //set to paused gif
+        var c = document.createElement('canvas');
+        var width = c.width = newImg.width = 290;
+        var height = c.height = newImg.height = 180;
+        var freeze = function(){
+          c.getContext('2d').drawImage(newImg,0,0,width,height);
+          c.style.position = 'absolute';
+          newImg.parentNode.insertBefore(c, newImg);
+        }
+        newImg.addEventListener('load', freeze, true);
+      })
+    } else if (result.newtab == "off"){
+      document.querySelectorAll(".newtab-link").forEach(function(link){
+        if(listener){
+          img.removeEventListener(onMouseOver);
+          img.removeEventListener(onMouseLeave);
+        }
+        //set thumbnail to favicon
+        var href = link.href;
+        link.querySelectorAll(".thumbnail").forEach(function(img){
+          var info = tippyTopSites.getSiteData(href);
+          if(info.image_url != null){
+            img.style.backgroundImage = "url(../node_modules/tippy-top-sites/images/"+info.image_url+")";
+            img.style.backgroundColor = info.background_color;
+            img.style.backgroundSize = "auto";
+            img.style.backgroundPosition = "center";
+          } else {
+            var newUrl = new URL("/favicon.ico", href);
+            img.style.backgroundImage = "url("+newUrl+")";
+            img.style.backgroundSize = "100px";
+            img.style.backgroundPosition = "center";
+          }
+        })
+      })
+    }
+  })
+}
+function onMouseOver(e){
+  //hide canvas/show gif
+  var i = 1.0;
+  var interval = setInterval(fade, 35);
+  function fade(){
+    e.target.style.opacity = i;
+    i-=0.1;
+    if (i <= 0){
+      clearInterval(interval);
+    }
+  }
+}
+function onMouseLeave(e){
+  //show canvas/hide gif
+  var i = 0.0;
+  var interval = setInterval(fade, 35);
+  function fade(){
+    e.target.style.opacity = i;
+    i+=0.1;
+    if (i >= 1.0){
+      clearInterval(interval);
+    }
+  }
+}
+function onStorageChange(changes, area) {
+  if(changes.newtab.newValue != changes.newtab.oldValue){
+    setThumbnail();
+    //refresh open new tab pages
+    var querying = browser.tabs.query({title: "New Tab"});
+    querying.then((tabs) => {
```

Comment:
Maybe we should only reload the tabs if they're `about:newtab` (or whatever our moz-extension newtab url is)?